src/main/relation/read_csv_relation.cpp

```diff
@@ -42,7 +42,7 @@ ReadCSVRelation::ReadCSVRelation(const shared_ptr<ClientContext> &context, const
 	CSVReaderOptions csv_options;
 	csv_options.file_path = file_name;
 	vector<string> empty;
-	csv_options.FromNamedParameters(options, *context);
+	context->RunFunctionInTransaction([&]() { csv_options.FromNamedParameters(options, *context); });
```

Comment:
Can we move the entire bind phase down here into a separate function, and run that entire function inside a transaction?

Starting a transaction is (relatively) heavy-handed, and it can now happen twice in this constructor. At that point I think it is better to run the entire code path within the same transaction.